src/read.c

```diff
@@ -1420,17 +1420,42 @@ static avifCodecType avifDecoderItemGetGridCodecType(const avifDecoderItem * gri
     return AVIF_CODEC_TYPE_UNKNOWN;
 }
 
-static avifResult avifDecoderGenerateImageGridTiles(avifDecoder * decoder, avifImageGrid * grid, avifDecoderItem * gridItem, avifItemCategory itemCategory)
+// Fills the dimgIdxToItemIdx array with a mapping from each 0-based tile index in the 'dimg' reference
+// to its corresponding 0-based index in the avifMeta::items array.
```

Comment:
IMPORTANT: Please check this question carefully because I am not familiar with this.

I see `avifDecoderItem::dimgIdx` documented as follows:

```
    // If dimgForId is non-zero, this is the zero-based index of this item in the list of Item #{dimgForID}'s dimg
    // Note that if the same item appears multiple times in the dimg box, this is the last index for this item.
    uint32_t dimgIdx;
```

Please verify that our code can handle the case where "the same item appears multiple times in the dimg box".

If I understand it correctly, in this case the `dimgIdxToItemIdx` array will have some entries left unset (equal to the special value `itemIndexNotSet`).

I wonder if `avifParseItemReferenceBox()` should store more information so that we can handle this case.

Reply:
See https://github.com/AOMediaCodec/libavif/pull/2315